modules/analysis/src/service.rs

```diff
@@ -117,14 +117,63 @@ pub fn ancestor_nodes(
     ancestor_nodes
 }
 
+pub async fn get_implicit_relationships(
+    connection: &ConnectionOrTransaction<'_>,
+    distinct_sbom_id: &str,
+) -> Result<Vec<QueryResult>, DbErr> {
+    // Retrieve all SBOM components that DO NOT have any defined relationships
+    let sql = format!(
+        r#"
+        SELECT
+             sbom.document_id,
+             sbom.sbom_id,
+             sbom.published::text,
+             get_purl(t1.qualified_purl_id) as purl,
+             t1_node.node_id,
+             t1_node.name AS node_name,
+             t1_version.version AS node_version,
+             product.name AS product_name,
+             product_version.version AS product_version
+        FROM
+            sbom
+        LEFT JOIN
+            product_version ON sbom.sbom_id = product_version.sbom_id
+        LEFT JOIN
+            product ON product_version.product_id = product.id
+        LEFT JOIN
+            sbom_node t1_node ON sbom.sbom_id = t1_node.sbom_id
+        LEFT JOIN
+            package_relates_to_package prtp ON t1_node.node_id = prtp.left_node_id OR t1_node.node_id = prtp.right_node_id
+        LEFT JOIN
+            sbom_package_purl_ref t1 ON t1_node.node_id = t1.node_id AND t1.sbom_id = sbom.sbom_id
+        LEFT JOIN
+            sbom_package t1_version ON t1_node.node_id = t1_version.node_id AND t1_version.sbom_id = sbom.sbom_id
+        WHERE
+            prtp.left_node_id IS NULL AND prtp.right_node_id IS NULL
+          AND
+            sbom.sbom_id = '{}'
```

Comment:
We should avoid this an do real SQL parameters.

Reply:
hehe, I thought this was 'real' sql params - you can teach me

yes thse changes are just for analysis graph.

Reply:
`format!` is just string concatenation. So a `'` might be problematic.

I think this is the function which would be the right one: https://docs.rs/sea-orm/latest/sea_orm/struct.Statement.html#method.from_sql_and_values